src/app/page.tsx

```diff
@@ -1,26 +1,59 @@
-import React from "react";
+"use client";
+
+import React, { useState, useEffect } from "react";
 import styles from "./index.module.css";
-import List from "../components/todo-list/TodoList";
+import TodoList from "../components/todo-list/TodoList";
 import { Input } from "@/components/input";
-import { Button } from "@/components/button";
+import { Button } from "../components/Button";
 
 async function getData() {
   const result = await fetch(
     "https://65c53ee5dae2304e92e41ae7.mockapi.io/api/todos/"
   );
-  // The return value is *not* serialized
-  // You can return Date, Map, Set, etc.
 
   if (!result.ok) {
     // This will activate the closest `error.js` Error Boundary
     throw new Error("Failed to fetch data");
   }
-
+  console.log(result);
   return result.json();
 }
 
-export default async function Home() {
-  const data = await getData();
+async function updateData(todoID: number, data: any) {
+  try {
+    const response = await fetch(
+      `https://65c53ee5dae2304e92e41ae7.mockapi.io/api/todos/${todoID}`,
```

Comment:
This is the second time you use `https://65c53ee5dae2304e92e41ae7.mockapi.io/api/todos/`. It would be a better to create a variable for url and use that one. This makes it easier to maintain in the future. For example when the url changes, you only need to change it at one spot.